core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -133,9 +143,7 @@ public void commit(TableMetadata base, TableMetadata metadata) {
             String.format("Update type %s is not supported", updateType));
     }
 
-    baseChanges.forEach(requestBuilder::update);
-    metadata.changes().forEach(requestBuilder::update);
-    UpdateTableRequest request = requestBuilder.build();
+    UpdateTableRequest request = new UpdateTableRequest(requirements, updates);
```

Comment:
The other request classes use a builder. I think that the intent here is to use the public constructor because the builder changed substantially and can't be updated to work for both. It would be nice to use the builder later, but I can see why this needs to be public and used at the moment.

Reply:
yes exactly, the builder changed substantially. In the long run I think we can add a builder again